web-app/src/hoc/Layout/index.js

```diff
@@ -6,10 +6,16 @@ import { LayoutContainer, Input } from './styled';
 
 export const Layout = props => {
   let drawer = null;
+
   if (props.isAuthenticated) {
     drawer = (
       <Fragment>
-        <Input type="checkbox" id="toggle-drawer" />
+        <Input
+          type="checkbox"
+          id="toggle-drawer"
+          onChange={props.toggleDrawer}
+          checked={props.drawerOpen}
```

Comment:
Probably good to destructure these props at the top plus readability for consistency across the app.